### PR TITLE
feat: add support for testnets

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -13,7 +13,8 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - id: set-matrix
-        run: echo "matrix=$(ls ./.github/tests/*.yaml | jq -R -s -c 'split("\n")[:-1]')" >> $GITHUB_OUTPUT
+        # List all yaml files in the .github/tests directory, except for the k8s.yaml file
+        run: echo "matrix=$(ls ./.github/tests/*.yaml | grep -vE 'k8s.yaml$' | jq -R -s -c 'split("\n")[:-1]')" >> $GITHUB_OUTPUT
```
Comment (h4ck3rk3y) [inline .github/workflows/nightly.yml:17]: is this chat gpt :P ?

File: src/participant_network.star [hunk 2]
```diff
@@ -149,14 +149,32 @@ def launch_participant_network(
...
             network_params.electra_fork_epoch,
         )
+    elif network_params.network in constants.PUBLIC_NETWORKS:
+        dummy = plan.run_sh(
+            run="mkdir /network-configs",
+            store=[StoreSpec(src="/network-configs/", name="el_cl_genesis_data")],
+        )
+        el_cl_data = el_cl_genesis_data.new_el_cl_genesis_data(
+            dummy.files_artifacts[0],
+            "0",
```
Comment (h4ck3rk3y) [inline src/participant_network.star:159]: lets clean up this magical constant

File: src/participant_network.star [hunk 3]
```diff
@@ -149,14 +149,32 @@ def launch_participant_network(
...
+        )
+
+        final_genesis_timestamp = 0
```
Comment (h4ck3rk3y) [inline src/participant_network.star:162]: do we not need these values for dora etc?

File: src/participant_network.star [hunk 4]
```diff
@@ -149,14 +149,32 @@ def launch_participant_network(
...
+        validator_data = None
     else:
         # split up the name from dencun-devnet-12 to dencun and devnet-12
-        devnet_name = network_params.network.split("-")[0]
-        devnet_number = network_params.network.split("-")[-1]
+        # split up the name from verkle-gen-devnet-12 to verkle-gen and devnet-12
+        network_parts = network_params.network.split("-devnet-", 1)
```
Comment (h4ck3rk3y) [inline src/participant_network.star:167]: can we take out the name splitting into a different function in the same file ( for readability )

File: src/participant_network.star [hunk 5]
```diff
@@ -149,14 +149,32 @@ def launch_participant_network(
...
+        devnet_name, devnet_number = network_parts[0], network_parts[1]
+        devnet_category = devnet_name.split("-")[0]
+        devnet_subname = (
+            devnet_name.split("-")[1] + "-" if len(devnet_name.split("-")) > 1 else ""
+        )
+        url = "github.com/ethpandaops/{0}-devnets/network-configs/{1}devnet-{2}".format(
```
Comment (h4ck3rk3y) [inline src/participant_network.star:173]: you can take the url format as a constant up top

File: src/shared_utils/shared_utils.star [hunk 6]
```diff
@@ -121,3 +121,27 @@ with open("/network-configs/network-configs/bootstrap_nodes.txt") as bootnode_fi
...
+
+def read_genesis_timestamp_from_config(plan, filename):
+    value = plan.run_python(
+        files={constants.GENESIS_DATA_MOUNTPOINT_ON_CLIENTS: filename},
+        wait=None,
+        run="""
+with open("/network-configs/config.yaml", "r") as f:
+    yaml_content = f.read()
+
+# Manually parse YAML-like content
```
Comment (h4ck3rk3y) [inline src/shared_utils/shared_utils.star:134]: WHY? why not use pyyaml?

Comment (h4ck3rk3y) [reply src/shared_utils/shared_utils.star:134]: this seems dangerous, lets use some out of the box yaml parser

**Review by h4ck3rk3y — APPROVED** [review]

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (barnabasbusa) [reply .github/workflows/nightly.yml:17]: yesgpt

File: src/shared_utils/shared_utils.star (hunk 6 above)
Comment (barnabasbusa) [reply src/shared_utils/shared_utils.star:134]: I tried to import yaml, but no packages were found.
wasn't sure how to import a package into run_python

File: src/participant_network.star (hunk 5 above)
Comment (barnabasbusa) [reply src/participant_network.star:173]: I would only want to "calculate" this url constant if I'm running a devnet

File: src/participant_network.star (hunk 3 above)
Comment (barnabasbusa) [reply src/participant_network.star:162]: Dora (and most other tools) have built in config for public testnets. 

I haven't bothered to do that yet, as didn't wanna get involved with gotemplating again.

File: src/participant_network.star (hunk 2 above)
Comment (barnabasbusa) [reply src/participant_network.star]: done

File: src/participant_network.star (hunk 3 above)
Comment (barnabasbusa) [reply src/participant_network.star]: Cleaned up too.

File: src/participant_network.star (hunk 4 above)
Comment (barnabasbusa) [reply src/participant_network.star]: Fixed

File: src/participant_network.star (hunk 5 above)
Comment (barnabasbusa) [reply src/participant_network.star]: Fixed

File: src/shared_utils/shared_utils.star (hunk 6 above)
Comment (barnabasbusa) [reply src/shared_utils/shared_utils.star]: fixed

**Pull request merged**